ui/components/MesheryPerformanceComponent.js

```diff
@@ -80,6 +80,13 @@ const styles = (theme) => ({
   },
 });
 
+const compulsoryProtocolValidUrlPattern = new RegExp('^(https?:\\/\\/)' // compulsory protocol
+  + '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' // domain name
+  + '((\\d{1,3}\.){3}\\d{1,3}))' // OR ip (v4) address
+  + '(\\:\\d+)?(\/[-a-z\\d%_.~+]*)*' // port and path
+  + '(\\?[;&a-z\\d%_.~+=-]*)?' // query string
+  + '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
+
```

Comment:
Great work @shubhamgupta2956 . Can you define the RegExp inside the handle change function, since it is only used there? Would be easier for future contributors to read.

Reply:
Sure @dhruv0000. But I had a suggestion. It may be possible that we may encounter other places in the future also where we may need this type of regex. Should we make a separate `constant.js/helper.js` file in which we can put these in one place? This will help when we are using the same function/constants in multiple files

Reply:
hmm. I don't think its needed for regex right now. But if ever needed, we have a lib folder for this kind of stuff.